src/components/MainContainer.js

```diff
@@ -1,22 +1,68 @@
-import React from "react";
-import StockContainer from "./StockContainer";
-import PortfolioContainer from "./PortfolioContainer";
-import SearchBar from "./SearchBar";
+import React, { useEffect, useState } from "react"
+import StockContainer from "./StockContainer"
+import PortfolioContainer from "./PortfolioContainer"
+import SearchBar from "./SearchBar"
 
 function MainContainer() {
+  const [stocks, setStocks] = useState(null)
+  const [portfolio, setPortfolio] = useState([])
+  const [filter, setFilter] = useState(null)
+  const [sort, setSort] = useState(null)
+
+  useEffect(() => {
+    fetch("http://localhost:3001/stocks")
+      .then((resp) => resp.json())
+      .then((data) => setStocks(data))
+  }, [])
+
+  const filteredStocks = filter
+    ? stocks.filter((stock) => stock.type === filter)
+    : stocks
+
+  const sortStocks = (stocks) => {
+    if (sort === "Alphabetically") {
+      return stocks.sort((a, b) => a.name.localeCompare(b.name))
+    } else if (sort === "Price") {
+      return stocks.sort((a, b) => a.price - b.price)
+    } else {
+      return stocks
+    }
+  }
```

Comment:
This should still work without the `else`. 
```suggestion
  const sortStocks = (stocks) => {
    if (sort === "Alphabetically") {
      return stocks.sort((a, b) => a.name.localeCompare(b.name))
    } else if (sort === "Price") {
      return stocks.sort((a, b) => a.price - b.price)
    }
  }
```